src/app/stream-deck/interfaces.ts

```diff
@@ -1,26 +1,200 @@
+  search: string;
+  page: string;
+  pullItems?: boolean;
+}
+
+// randomize the current character
+// both modes (weapon only / all)
+export interface RandomizeAction {
+  action: 'randomize';
+  weaponsOnly: boolean;
+}
+
+// collect all items from postmaster
+export interface CollectPostmasterAction {
+  action: 'collectPostmaster';
+}
+
+// trigger refresh DIM
+export interface RefreshAction {
+  action: 'refresh';
+}
+
+// enable/disable farming mode
+export interface FarmingModeAction {
+  action: 'farmingMode';
+}
+
+// maximize power
+export interface MaxPowerAction {
+  action: 'maxPower';
+}
+
+// pick a random item of the selected bucket
+// and move it to the vault to free a slot
+export interface FreeBucketSlotAction {
+  action: 'freeBucketSlot';
+  bucket: InventoryBucket['type'];
+}
+
+// pull a selected item from other character/vault
+// (if the current character has already that item it will be moved to the vault)
+export interface PullItemAction {
+  action: 'pullItem';
+  item: string;
+}
+
+// allow the user to pick a specific "thing" and send it to the Stream Deck
+// this thing can be a loadout or an item
+export interface SelectionAction {
+  action: 'selection';
+  selection: StreamDeckSelectionType;
+}
+
+// equip a selected loadout (for a specific store)
+// send the shareable link of a loadout to the Stream Deck
+export interface EquipOrShareLoadoutAction {
+  action: 'loadout' | 'shareLoadout';
+  loadout: string;
+  character: string;
+}
+
+export interface AuthorizationInitAction {
+  action: 'authorization:init';
+}
+
+export interface AuthorizationConfirmAction {
+  action: 'authorization:confirm';
+  challenge?: number;
+}
+
+export type StreamDeckMessage = (
+  | AuthorizationInitAction
+  | AuthorizationConfirmAction
+  | SearchAction
+  | RandomizeAction
+  | CollectPostmasterAction
+  | RefreshAction
+  | FarmingModeAction
+  | MaxPowerAction
+  | FreeBucketSlotAction
+  | PullItemAction
+  | SelectionAction
+  | EquipOrShareLoadoutAction
+) & { token?: string };
+
+// Types of messages sent to Stream Deck
+
+export interface VaultArgs {
+  vault: number;
+  shards?: number;
+  glimmer?: number;
+  brightDust?: number;
+}
+
+export interface MetricsArgs {
+  gambit: number;
+  vanguard: number;
+  crucible: number;
+  trials: number;
+  gunsmith: number;
+  ironBanner: number;
+  triumphs: number;
+  battlePass: number;
+  artifactIcon?: string;
+}
+
+export interface PostmasterArgs {
+  total: number;
+  ascendantShards: number;
+  enhancementPrisms: number;
+  spoils: number;
+}
+
+export interface MaxPowerArgs {
+  artifact: number;
+  base: string;
+  total: string;
+}
+
+export interface Challenge {
+  label: number;
+  value: string;
+}
+
+export interface SendToStreamDeckArgs {
+  // action to execute on Stream Deck
+  action: 'dim:update' | 'dim:share-url' | 'authorization:challenges' | 'authorization:reset';
+  // AuthorizationNotification token
+  token?: string;
+  // actions parameters
```

Comment:
Why wouldn't messages sent to the StreamDeck follow the same pattern of being a union of a type and an argument, rather than this union of all possible arguments?